pkg/executor/util.go

```diff
@@ -24,6 +24,21 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// Service names size have a limitation of max 63 characters. The service
+// names are derived from statefulset names that have a 5 character UID
+// appended towards the end. While calculating the max prefix size for the
+// service names, the 5 digit UID and the 4 digit maxKDMember size (1000)
+// should be accounted for.
+// Also, as part of stateful pod creating a 10 digit hash value is added
+// to the controller revision hash label which needs to be accounted for
+// while calculating the prefix size.
+// Naming scheme for the service is as follows: prefix + UID + member index
+// Naming scheme for the label is as follows: prefix + UID + hash value
+// Since, the max member size currently is restricted to be 4 characters, take
+// the max of hash value digits and member size digits which is 10.
+// Prefix calculation is done as following = 63 - 10 - 5 - 2 ('-' characters) = 46.
```

Comment:
Good explanation! except only one hyphen character will be added, not 2. The first hyphen is part of the prefix whose length we are enforcing.

Also, this calculation determines the maximum valid size of the prefix, i.e. a value that the prefix can be <= to. However the code that references this value enforces that the prefix is < the value rather than <=.